core/src/execution/proto/operator.proto

```diff
@@ -87,3 +88,21 @@ message Expand {
   repeated spark.spark_expression.Expr project_list = 1;
   int32 num_expr_per_project = 3;
 }
+
+message HashJoin {
```

Comment:
can we name it as `Join` so that we can use it for both SMJ and SHJ / BHJ and I will further use it in BNLJ change I am working on ?

Reply:
They are different join operators. I'm not sure how we use same `Join` to represent them?

Reply:
I was thinking of something like this :
```
message Join {
  repeated spark.spark_expression.Expr left_join_keys = 1;
  repeated spark.spark_expression.Expr right_join_keys = 2;
  JoinType join_type = 3;
  // can serve as condition in SHJ and sort_options in SMJ
  repeated spark.spark_expression.Expr join_exprs = 4;
  JoinExec join_exec = 5;
}

message JoinExec {
 HashJoin = 0;
 SMJ = 1;
....
}
```

may be it's too much and having diff proto msg for each join should be right thing to do !